lib/src/calendar_event_data.dart

```diff
@@ -69,7 +69,16 @@ class CalendarEventData<T extends Object?> {
     return other is CalendarEventData<T> &&
         date.compareWithoutTime(other.date) &&
         endDate.compareWithoutTime(other.endDate) &&
-        event == other.event &&
+        ((event == null && other.event == null) ||
+            (event != null && other.event != null && event == other.event)) &&
```

Comment:
what is the reason for checking == null as well as != null for the same parameter?

Reply:
Basically event is nullable. So, logic here is if both events are null or both are non null and same objects then we can consider it as same. If not then they are not same object.